addon/components/mobile/flexberry-objectlistview.js

```diff
@@ -138,4 +140,132 @@ export default FlexberryObjectlistview.extend({
     @default true
   */
   columnsWidthAutoresize: true,
+
+  /**
+    Array of objects corresponding to list of pages.
+
+    Each page is presented as object with following properties:
+    - **number** - Number of page.
+    - **isCurrent** - If `true` this page is current.
+    - **isShow** - If `true` this page showing in pages list.
+
+    @property mobilePages
+    @type Array
+    @readOnly
+  */
+  mobilePages: Ember.computed('pages', function() {
+    let mobilePages = Ember.A();
+    let pages = this.get('pages');
+    let currentPageNumber = pages.find(page => page.isCurrent).number;
+    let lastPageNumber = pages[pages.length - 1].number;
+
+    for (let i = 1; i <= lastPageNumber; i++) {
+      let isShow;
+
+      if (currentPageNumber === 1 || currentPageNumber === lastPageNumber) {
+        isShow  = currentPageNumber === 1 ? i <= 3 : i >= lastPageNumber - 2;
+      } else {
+        isShow = currentPageNumber - 1 <= i && i <= currentPageNumber + 1;
+      }
+
+      let page = {
+        isCurrent: i === currentPageNumber,
+        isShow: isShow,
+        number: i
+      };
+
+      mobilePages.pushObject(page);
+    }
+
+    return mobilePages;
+  }),
+
+  /*
+    Convert array of object sorting to array.
+
+    @private
+    @property _currecntSortingArray
+    @type Array
+    @readOnly
+  */
+  _currecntSortingArray: Ember.computed('sorting',  function() {
+    let sorting = this.get('sorting');
+    let sortingKeys = Object.keys(sorting);
+    let columns = Ember.A();
+    sortingKeys.forEach(key => {
+      let column = sorting[key];
+      columns.pushObject({
+        key: key,
+        sortNumber: column.sortNumber,
+        sortAscending: column.sortAscending
+      });
+    });
+
+    columns.sortBy('sortNumber');
+
+    return columns;
+  }),
+
+  /**
+    Class icons for sorting.
+
+    @property mobileSortingSettingsIcon
+    @type String
+    @readOnly
+  */
+  mobileSortingSettingsIcon: Ember.computed('sorting',  function() {
```

Comment:
Это свойство должно зависеть от `_currecntSortingArray`, и его тоже лучше приватным сделать.
```suggestion
    @private
    @property _mobileSortingSettingsIcon
    @type String
    @readOnly
  */
  _mobileSortingSettingsIcon: Ember.computed('_currecntSortingArray',  function() {
```